src/webrtc/call.ts

```diff
@@ -1300,31 +1504,48 @@ export class MatrixCall extends EventEmitter {
             return;
         }
 
-        const oldRemoteStream = this.feeds.find((feed) => {return !feed.isLocal()})?.stream;
+        const stream = ev.streams[0];
+        this.pushRemoteFeed(stream);
+        stream.addEventListener("removetrack", () => this.deleteFeedByStream(stream));
+    };
 
-        // If we already have a stream, check this track is from the same one
-        // Note that we check by ID and always set the remote stream: Chrome appears
-        // to make new stream objects when tranciever directionality is changed and the 'active'
-        // status of streams change - Dave
-        if (oldRemoteStream && ev.streams[0].id !== oldRemoteStream.id) {
-            logger.warn(
-                `Ignoring new stream ID ${ev.streams[0].id}: we already have stream ID ${oldRemoteStream.id}`,
-            );
-            return;
-        }
+    /**
+     * This method removes all video/rtx codecs from video transceivers.
+     * This is necessary since they can cause problems. Without this the
+     * following steps should produce an error:
+     *   Chromium calls Firefox
+     *   Firefox answers
+     *   Firefox starts screen-sharing
+     *   Chromium starts screen-sharing
+     *   Call crashes for Chromium with:
+     *       [96685:23:0518/162603.933321:ERROR:webrtc_video_engine.cc(3296)] RTX codec (PT=97) mapped to PT=96 which is not in the codec list.
+     *       [96685:23:0518/162603.933377:ERROR:webrtc_video_engine.cc(1171)] GetChangedRecvParameters called without any video codecs.
+     *       [96685:23:0518/162603.933430:ERROR:sdp_offer_answer.cc(4302)] Failed to set local video description recv parameters for m-section with mid='2'. (INVALID_PARAMETER)
+     */
+    private getRidOfRTXCodecs() {
+        // RTCRtpReceiver.getCapabilities and RTCRtpSender.getCapabilities don't seem to be supported on FF
+        if (RTCRtpReceiver.getCapabilities && RTCRtpSender.getCapabilities) {
+            const recvCodecs = RTCRtpReceiver.getCapabilities("video").codecs;
+            const sendCodecs = RTCRtpSender.getCapabilities("video").codecs;
+            const codecs = [...sendCodecs, ...recvCodecs];
+
+            for (const codec of codecs) {
+                if (codec.mimeType === "video/rtx") {
+                    const rtxCodecIndex = codecs.indexOf(codec);
+                    codecs.splice(rtxCodecIndex, 1);
+                }
+            }
 
-        if (!oldRemoteStream) {
-            logger.info("Got remote stream with id " + ev.streams[0].id);
+            for (const trans of this.peerConn.getTransceivers()) {
+                if (
+                    trans.sender.track?.kind === "video" ||
+                    trans.receiver.track?.kind === "video"
+                ) {
+                    trans.setCodecPreferences(codecs);
+                }
+            }
```

Comment:
Good point - I'll look into it once I have time. That might be a bit longer than usual because of school and some other stuff, sorry...

Reply:
I ended up getting a bit of time:

> Also, is there a reason not to get the capabilities from each transceivers' sender

The [docs](https://developer.mozilla.org/en-US/docs/Web/API/RTCRtpSender) say it is only a static method and so does VS Code.

> Also if we're doing this, I think we need to at least refer to some upstream bug that justifies the workaround.

In that case, I would need to get a repro case. I'd like to do that but it might take quite some time...

Reply:
Thanks - we should still chase this up at some point but if we can just turn RTX off for the screenshare stream then it's less of a priority.

Reply:
Definitely agreed - it's on my TODO list